status-server/src/index.js

```diff
@@ -42,26 +40,35 @@ app.get('/'+config.prefix+':num\.:ext?', function (req, res) {
         return res.status(500).send('Error reading status last-modified!');
       }
       res.setHeader('Last-Modified', stats.mtime);
-      switch (ext.toLowerCase()) {
-        case 'json':
-          res.json({ 'last-modified': (new Date(stats.mtime)).getTime(), value: status });
-          break;
-        case 'html':
-          res.set('Content-Type', 'text/html');
-          res.send( htmlResponse({ url: req.path, name: 'status', value: status }));
-          break;
-        default:
-          res.send(status);
+      res.json({ 'last-modified': (new Date(stats.mtime)).getTime(), value: status });
+    });
+  });
+});
+//handles html requests based on :id, the status number. reads file of :id provided and renders html page with an update bar to update status
+app.get('/user/:id/status', function (req, res) {
+  var num = req.params.id || 0;
+  var filename = path.join(dataDir, 'status' + num);
+  fs.readFile(filename, function(err, buf) {
+    if (err) {
+      console.error(err.stack);
+      return res.status(500).send('Error reading status!');
+    }
+    var status = buf.toString()
+    fs.stat(filename, function(err, stats) {
+      if (err) {
+        console.error(err.stack);
+        return res.status(500).send('Error reading status last-modified!');
       }
+      res.setHeader('Last-Modified', stats.mtime);
+      res.set('Content-Type', 'text/html');
+      res.send( htmlResponse({ url: req.path, name: 'status', value: status }));
```

Comment:
lets break up 'url' here into its constituent parts: the prefix ('api'), the user id and the resource/path